### PR TITLE
NBS-1526 [disk manager] call mlock on dm process start

### PR DESCRIPTION
**Pull request by gy2411**
Lock process memory ranges containing binary and .so files when process starts. Similar to nbs: https://github.com/ydb-platform/nbs/blob/main/cloud/storage/core/libs/daemon/mlock.cpp

After memory is locked, /proc/<pid>/status and /proc/<pid>/maps for dm process look like this:
[dm_memory_state.txt](https://github.com/ydb-platform/nbs/files/15027268/dm_memory_state.txt) 
As we see, 39572 kB are locked.

NOTE! Before starting dm with this change, LimitMEMLOCK parameter in [Service] section of `/usr/lib/systemd/system/yc-disk-manager.service` should be set. Otherwise the process will fail to lock memory and disk manager will not start. I suggest `LimitMEMLOCK=268435456` (256 MiB).

### CODE REVIEW

File: cloud/disk_manager/internal/pkg/util/mlock.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+
+	"github.com/ydb-platform/nbs/cloud/tasks/errors"
+)
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/util/mlock.go:11]: ////...

like here https://github.com/ydb-platform/nbs/blob/main/cloud/disk_manager/pkg/app/controlplane.go#L37

File: cloud/disk_manager/internal/pkg/util/mlock_test.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
+package util
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/util/mlock_test.go:7]: ///...

File: cloud/disk_manager/internal/pkg/util/mlock_test.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+	for _, testCase := range cases {
+		item, err := parseProcMapsLine(testCase.line)
+		require.NoError(t, err)
+		shouldLock := shouldLockRange(item)
+		require.EqualValues(t, testCase.shouldLockExpected, shouldLock)
+	}
+}
+
+func TestLockBinaryDoesNotFail(t *testing.T) {
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/util/mlock_test.go:96]: TestLockProcessMemory

File: cloud/disk_manager/internal/pkg/util/mlock_test.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+	for _, testCase := range cases {
+		item, err := parseProcMapsLine(testCase.line)
+		require.NoError(t, err)
+		require.EqualValues(t, testCase.itemExpected, *item)
+	}
+
+	_, err := parseProcMapsLine("02443000-02444000 r--p 02242000 fd:11")
+	require.Error(t, err)
+
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/util/mlock_test.go:37]: unnecessary \n

File: cloud/disk_manager/internal/pkg/util/mlock.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		"%x-%x %s %x %s %d %s",
+		&item.memRange.start,
+		&item.memRange.end,
+		&item.permissions,
+		&item.offset,
+		&item.device,
+		&item.inode,
+		&item.pathname,
+	)
+	if err != nil && (err != io.EOF || n < 6) {
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/util/mlock.go:40]: Why checking io.EOF is needed here?

Do we have tests where io.EOF happens while scanning here?

Comment (SvartMetal) [reply cloud/disk_manager/internal/pkg/util/mlock.go:42]: Can we separate this if into some, and write comment about every case?

**Review by SvartMetal — APPROVED** [review]

**Review by BarkovBG — APPROVED** [review]

**Pull request merged**